connectivity/FEATURE_BLE/source/cordio/source/GattServerImpl.cpp

```diff
@@ -588,6 +588,23 @@ ble_error_t GattServer::insert_cccd(
     cccds[cccd_cnt].secLevel = characteristic->getUpdateSecurityRequirement().value();
     cccd_handles[cccd_cnt] = characteristic->getValueAttribute().getHandle();
 
+    /**
+     * Set the characteristic's implicitly-created CCCD
+     *
+     * Ownership is passed to the GattCharacteristic
+     */
+    GattAttribute* implicit_cccd = new GattAttribute(
```

Comment:
Can you create the attribute with `std::nothrow` and return an error if it fails ? 
If the code stays at this position `currentHandle` should be decremented before return.

Reply:
Please see f9af08c and check if the change is sufficient